packages/jest-snapshot/src/__tests__/utils-test.js

```diff
@@ -30,3 +36,17 @@ test('getSnapshotPath()', () => {
     path.join('/abc', 'cde', '__snapshots__', 'a-test.js.snap'),
   );
 });
+
+test('saveSnapshotFile()', () => {
+  const data = {
+    myKey: '<div>\r\n</div>', 
+  };
+
+  const file = path.join(__dirname, '.tmp.snapshot');
+  saveSnapshotFile(data, file);
+
+  const content = fs.readFileSync(file, 'utf-8');
+  expect(content).toBe('exports[`myKey`] = `<div>\n</div>`;\n');
+
+  fs.unlinkSync(file);
```

Comment:
we need to unlink in `afterEach` instead (using try-catch to make sure it doesn't throw).